crates/api_models/src/payment_methods.rs

```diff
@@ -901,7 +904,37 @@ pub struct ConnectorTokenDetails {
     pub metadata: Option<pii::SecretSerdeValue>,
 
     /// The value of the connector token. This token can be used to make merchant initiated payments ( MIT ), directly with the connector.
-    pub token: String,
+    pub token: masking::Secret<String>,
+}
+
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+impl ConnectorTokenDetails {
+    // show only first and last two digits of the key and mask others with *
+    // mask the entire key if it's length is less than or equal to 4
+    fn mask_value(&self, val: String) -> masking::Secret<String> {
+        let val_len = val.len();
+        let masked_val = if val_len <= 4 {
+            "*".repeat(val_len)
+        } else {
+            // Show the first two and last two characters, mask the rest with '*'
+            let mut masked_val = String::new();
+            let val_len = val.len();
+            // Iterate through characters by their index
+            for (index, character) in val.chars().enumerate() {
+                if index < 2 || index >= val_len - 2 {
+                    masked_val.push(character); // Keep the first two and last two characters
+                } else {
+                    masked_val.push('*'); // Mask the middle characters
+                }
+            }
+            masked_val
+        };
+        masking::Secret::new(masked_val)
+    }
+
+    pub fn set_masked_token(&mut self) {
```

Comment:
This suffers from the same concerns pointed out in https://github.com/juspay/hyperswitch/pull/6798#discussion_r1880781752, and allows us to construct a `ConnectorTokenDetails` without a masked value for the `token` field.

If providing a masked token in the payment method response is still intentional, then can we ensure that the `token` field would always be masked? 

Maybe we can define a newtype like this, which would be used for the `token` field?
https://github.com/juspay/hyperswitch/blob/d1f53036c75771d8387a9579a544c1e2b3c17353/crates/common_utils/src/new_type.rs#L78-L91